src/tier/tier.service.ts

```
@@ -0,0 +1,41 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { Features, lookupOrg, subscribe } from "tier";
+
+@Injectable()
+export class TierService {
+  private logger = new Logger(TierService.name);
+
+  constructor(private configService: ConfigService) {}
+
+  async checkAddOrg(userId: number, userDetails: { email: string; name: string; login: string }, proAccount = false) {
+    try {
+      // check to see if the user is registered
+      await lookupOrg(`org:${userId}`);
+    } catch (e) {
+      const { email, name, login } = userDetails;
+
+      // grandparent existing pro account if necessary
+      const plan: Features = (
+        proAccount ? this.configService.get("tier.proPlan")! : this.configService.get("tier.freePlan")!
+      ) as Features;
+
+      try {
+        // register the user with a free/pro account
+        await subscribe(`org:${userId}`, [plan] as Features[], {
+          info: {
+            email,
+            name,
+            description: `OpenSauced Free Account`,
+            phone: "",
+            metadata: {
+              login,
+            },
+          },
+        });
+      } catch (e) {
+        this.logger.error(`Unable to register account for ${login}`);
+      }
```

Do we want to surface the error here? How we've done error handling in the past has confused me abit due to typescripts gnarly handling of errors but elsewhere I've seen us do:

https://github.com/open-sauced/api/blob/692f953192dff8449f3716a0a28c88f20f1f741c/src/blogs/blog-summary.service.ts#L31-L34

---

Yep, good catch